app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
                 }
 
                 bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
+                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
+
+                bgReading.save();
+                bgReading.perform_calculations();
+                Notifications.getInstance(context).notificationSetter(context);
+                BgSendQueue.addToQueue(bgReading, "create", context);
+            }
+        }
+        Log.w("BG GSON: ",bgReading.toS());
+
+        return bgReading;
+    }
+    public static BgReading create(double raw_data, Context context, Long timestamp) {
```

Comment:
It isn't clear in the code that the existing function saves the filtered_data & raw_data, especially by the function parameters. This function ensures the dexbridge data is stored fully without upsetting the existing code. 

 
John Stevens
M - +61 418526349

> On 19 May 2015, at 23:56, Stephen Black <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java:
> 
> > @@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
> >                  }
> >  
> >                  bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
> > +                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
> > +
> > +                bgReading.save();
> > +                bgReading.perform_calculations();
> > +                Notifications.getInstance(context).notificationSetter(context);
> > +                BgSendQueue.addToQueue(bgReading, "create", context);
> > +            }
> > +        }
> > +        Log.w("BG GSON: ",bgReading.toS());
> > +
> > +        return bgReading;
> > +    }
> > +    public static BgReading create(double raw_data, Context context, Long timestamp) {
> Im not sure what the purpose of this method is, it the two were already handled in the other branch
> an option if you really wanted to keep them separate would be
> 
> public static BgReading create(double raw_data, Context context, Long timestamp) {
>   BgReading.create(raw_data, raw_data, context, timestamp);
> }
> it will just map this function to the existing one
> 
> —
> Reply to this email directly or view it on GitHub.
>